src/framework/theme/components/tree-grid/data-source/tree-grid-data.service.ts

```diff
@@ -36,14 +48,11 @@ export class NbTreeGridDataService<T> {
 
   copy(nodes: NbTreeGridPresentationNode<T>[]): NbTreeGridPresentationNode<T>[] {
     return nodes.map((node: NbTreeGridPresentationNode<T>) => {
-      const presentationNode = new NbTreeGridPresentationNode(node.node, node.level);
-      presentationNode.expanded = node.expanded;
-
-      if (node.hasChildren()) {
-        presentationNode.children = this.copy(node.children);
-      }
+      const children: NbTreeGridPresentationNode<T>[] = node.hasChildren()
```

Comment:
The same thing here:
```suggestion
      const children: NbTreeGridPresentationNode<T>[] = node.hasChildren() && this.copy(node.children);
```

Why don't you like this style? It's not first I'm noticing you're using ternary operators instead of plain *&&*. Why do you decide that way?)

Reply:
In case when `node.hasChildren()` is falsey value, children would become this value instead of `undefined`. Personally, I don't like this behavior, because if my property defined as `{} | undefined` I can use `value === undefined` to verify it's not defined. It won't work with other falsey values 🙂

Reply:
Ok, then, let's use plain *if/else* statement. I think it's the best solution 😄

Reply:
Yep, did like this already. Thx